addons/mail/static/src/core/common/thread_actions.js

```diff
@@ -58,7 +90,8 @@ function transformAction(component, id, action) {
         },
         /** Name of this action, displayed to the user. */
         get name() {
-            return this.isActive && action.nameActive ? action.nameActive : action.name;
+            const res = this.isActive && action.nameActive ? action.nameActive : action.name;
+            return typeof res === "function" ? res(component) : res;
```

Comment:
why whould the name be a function?

Reply:
because it returns the text based on the specific status of the chat window. I also agree that this way of coding can be a little bit confusing.